Library/Homebrew/command.rb

```diff
@@ -0,0 +1,157 @@
+module Homebrew
+  class Command
+    attr_reader :command_name, :valid_options, :description, :help_output,
+      :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @root_options = []
+      @argv = ARGV.dup
+    end
+
+    def options(&block)
+      initialize_variables
+      @parent = nil
+      instance_eval(&block)
+      generate_help_and_manpage_output
+    end
+
+    def cmd_name(cmd)
+      @command_name = cmd
+    end
+
+    def add_valid_option(option_hash)
```

Comment:
I'd just inline this method as `valid_options.push option_hash` at the one call site